### PR TITLE
Fix missing permissions on newer Android versions

### LINKED ISSUE #270: Notifications and reconnecting to WebSocket not working on newer Android versions
Because of the build and sdk updates, we need to request additional permissions now.
Fix is coming very soon

### PR DESCRIPTION
**Pull request by cyb3rko**
I have decided to use [QuickPermissions Kotlin](https://github.com/QuickPermissions/QuickPermissions-Kotlin) as it offers a very simple solution for permission handling.

If you say you don't want that additional library we can surely find a solution.
This PR should bring back full functionality to the Kotlin version of the app.

Closes #270

### CODE REVIEW

**Comment by cyb3rko** [comment]
Note:
The app requests notification AND exact alarm permission, but atm requesting the latter does not do anything.
Maybe that will change in future Android versions.

File: app/src/main/kotlin/com/github/gotify/init/InitializationActivity.kt [hunk 1]
```diff
@@ -130,4 +134,23 @@ internal class InitializationActivity : AppCompatActivity() {
             .version
             .enqueue(Callback.callInUI(this, callback, errorCallback))
     }
+
+    private fun runWithNeededPermissions(action: () -> Unit) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) {
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+                // Android 13 and above
+                runWithPermissions(
```
Comment (jmattheis) [inline app/src/main/kotlin/com/github/gotify/init/InitializationActivity.kt:142]: Currently, if you deny everything, the app only shows the initialization scree. I think the prompt shouldn't be cancelable. As we require these permissions.
image

File: app/build.gradle [hunk 2]
```diff
@@ -68,6 +68,7 @@ dependencies {
     implementation 'androidx.vectordrawable:vectordrawable:1.1.0'
     implementation 'androidx.preference:preference:1.2.0'
 
+    implementation 'com.github.quickpermissions:quickpermissions-kotlin:0.4.0'
```
Comment (jmattheis) [inline app/build.gradle:71]: The library hasn't received an update since the end of 2019 :/

**Review by jmattheis — CHANGES_REQUESTED** [review]

File: app/build.gradle (hunk 2 above)
Comment (cyb3rko) [reply app/build.gradle:71]: Don't worry, I've forked it and updated everything:
https://github.com/cyb3rko/QuickPermissions-Kotlin

File: app/src/main/kotlin/com/github/gotify/init/InitializationActivity.kt (hunk 1 above)
Comment (cyb3rko) [reply app/src/main/kotlin/com/github/gotify/init/InitializationActivity.kt:151]: Fixed.

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -45,6 +45,9 @@
     <string name="select_ca_failed">Failed to read CA: %s</string>
     <string name="login">Login</string>
     <string name="check_url">Check URL</string>
+    <string name="permissions_dialog_grant">Grant</string>
+    <string name="permissions_denied_temp">Gotify requires permission to display push notifications.</string>
```
Comment (jmattheis) [inline app/src/main/res/values/strings.xml:49]: I've made this a little more specific.

File: app/build.gradle (hunk 2 above)
Comment (jmattheis) [reply app/build.gradle]: I've looked through the repo, nicely done!

**Review by jmattheis — APPROVED** [review]

File: app/src/main/res/values/strings.xml (hunk 3 above)
Comment (cyb3rko) [reply app/src/main/res/values/strings.xml:49]: To be exact it requires the exact alarm permission as well but you don't see a permission dialog atm as it given right away.
But yeah, possible for now.

**Pull request merged**